src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -194,8 +195,25 @@ void ov::npuw::util::unpack(const ov::SoPtr<ov::ITensor>& from,
             NPUW_ASSERT(false);
         }
     } else if (type_from == ov::element::u8) {
-        // Only support CW for now
-        if (scale_shape.size() == 2 && scale_shape[0] == from_shape[0] && scale_shape[1] == 1) {
+        if (scale_shape.size() == 3 && scale_shape[1] == 1 && scale_shape[2] == 1) {
+            // Special case for broadcasting vocab by 2 dimensions
+            // FIXME: all this logic probably should be in some specific unpack or another util function
+            ov::Tensor wraped_from(from->get_element_type(),
+                                   ov::Shape{from_shape[0], from_shape[1] * from_shape[2]},
+                                   from->data());
+            ov::Tensor wraped_zerop(zerop->get_element_type(),
+                                    ov::Shape{zerop_shape[0], zerop_shape[1] * zerop_shape[2]},
+                                    zerop->data());
+            ov::Tensor wraped_scale(scale->get_element_type(),
+                                    ov::Shape{scale_shape[0], scale_shape[1] * scale_shape[2]},
+                                    scale->data());
```

Comment:
you've missed strides here

Reply:
Original strides don't work - the shape's changed. I cannot just multiply them. How to properly calculate them? Also it works just fine w/o any strides

Reply:
You need to calculate new ones from old ones.

The fact it worked just means buffers are continuous but it may not always be the case.

Reply:
So.. Strides are distances in bytes between dimensions.

You represent a 3D tensor with strides `(s2, s1, s0)` as s 2D tensor with strides `(s'1, s'0)`.

`s0` is a distance between each element in a tensor in the innermost dimension. Normally it is `sizeof(T)` though I am not sure what's the case for `u4`/`i4` types - but you can print those out for your tensors.

For you it doesn't change.

`s'1` is a distance between each row (of elements). That is, in your case it is `(scale_shape[1]*scale_shape[2]) * s'0`.

Reply:
Done